TV by the Numbers.js

```diff
@@ -1,144 +1,208 @@
+	translator.setDocument(doc);
 
-	while (post = posts.iterateNext()) {
-		var newItem = new Zotero.Item("webpage");
-		newItem.publicationTitle = PUB_TITLE;
-		
-		var link = post.getElementsByTagName("a")[0];
-		newItem.url = link.href;
-		
-		var title = Zotero.Utilities.unescapeHTML(
-			Zotero.Utilities.cleanTags(link.textContent));
-		title = title.replace(/(\s+)(?:‘|’)|(?:‘|’)(\s+)/g, "$1''$2").replace(/‘|’/g, "'");
-		newItem.title = title;
-		
-		if (XPATH_DATE)
-			newItem.date = xpre(doc, post, XPATH_DATE, RE_DATE);
-		if (XPATH_PAGES)
-			newItem.pages = xpath_string(doc, post, XPATH_PAGES);
-		
-		//authors
-		var author_text = xpre(doc, post, XPATH_AUTHORS, RE_AUTHORS);
-		var authors = [];
-		if (author_text) {
-			if (author_text.indexOf(" and ") != -1)
-				authors = author_text.split(" and ");
-			else if (author_text.indexOf(";") != -1)
-				authors = author_text.split(";");
+	translator.setHandler('itemDone', function(obj, item) {
+		item.date = ZU.strToISO(item.date);
+		var authors = ZU.xpath(doc, '//a[@rel="author"]');
+		for (var i in authors) {
+			if (authors[i].textContent == 'TV By The Numbers')
+				item.creators.push({
+					lastName: authors[i].textContent,
+					creatorType: "author",
+					fieldMode: 1
+				});
 			else
-				authors.push(author_text);
-		}
-		for (var i=0; i<authors.length; i++) {
-			if (authors[i] != 'null') {
-				newItem.creators.push(
-					Zotero.Utilities.cleanAuthor(authors[i], "author"));
-			}
+				item.creators.push(ZU.cleanAuthor(authors[i].textContent, "author"));
 		}
+		item.complete();
+	});
 
-		// attach html
-		if (url == newItem.url)
-			newItem.attachments.push({title:PUB_TITLE+" Snapshot",
-				mimeType:"text/html", url:doc.location.href, snapshot:true});
-		
-		newItem.toString = function() { return this.title; };
-		items[newItem.url] = newItem;
-		post_count++;
+	translator.getTranslatorObject(function(trans) {
+		trans.itemType = "webpage";
+		trans.doWeb(doc, url);
+	});
+} /** BEGIN TEST CASES **/
+var testCases = [{
+		"type": "web",
+		"url": "http://tvbythenumbers.zap2it.com/page/20/?s=harry+potter",
+		"items": "multiple"
+	},
+	{
+		"type": "web",
+		"url": "http://tvbythenumbers.zap2it.com/daily-ratings/monday-final-ratings-july-3-2017/",
+		"items": [{
+			"itemType": "webpage",
+			"title": "‘American Ninja Warrior’ adjusts up: Monday final ratings",
+			"creators": [{
+				"firstName": "Rick",
+				"lastName": "Porter",
+				"creatorType": "author"
+			}],
+			"date": "2017-07-06",
+			"abstractNote": "Final broadcast primetime live + same-day ratings for Monday, July 3, 2017 The top show on a rerun-filled Monday night saw its adults 18-49 rating grow from the preliminary numbers to the finals. &…",
+			"shortTitle": "‘American Ninja Warrior’ adjusts up",
+			"url": "http://tvbythenumbers.zap2it.com/daily-ratings/monday-final-ratings-july-3-2017/",
+			"websiteTitle": "TV By The Numbers by zap2it.com",
```

Comment:
I'd set the website title in the scraper so that it reads "TV By The Numbers" (not the name EM found)